source/documentation.html.md

```diff
@@ -40,436 +40,2029 @@ class DiscordClient {
+
+Uses [Modify Current User Nick](https://discordapp.com/developers/docs/resources/guild#modify-current-user-nick)
+
+### addRole
+
+```cpp
+bool addRole(Snowflake<Server> serverID, Snowflake<User> userID, Snowflake<Role> roleID);
+```
+
+Gives a member a role on a server
+
+#### Parameters
+<table>
+  <tbody>
+      <tr><td><strong>serverID</strong></td>
+        <td>The id of the server with the user you want to give the role to</td></tr>
+      <tr><td><strong>memberID</strong></td>
+        <td>The id of the user that you want to give the role to</td></tr>
+      <tr><td><strong>roleID</strong></td>
+        <td>The id of role that you want to give</td></tr>
+  </tbody>
+</table>
+
+#### Return value
+Returns ``true`` on success
+
+#### Other Details
+[Declared in `client.h`](https://github.com/NoNamer64/sleepy-discord/blob/master/include/sleepy_discord/client.h) and [defined in `endpoints.cpp`](https://github.com/NoNamer64/sleepy-discord/blob/master/include/sleepy_discord/endpoints.cpp)
+
+Uses [Add Guild Member Role](https://discordapp.com/developers/docs/resources/guild#add-guild-member-role)
+
+### removeRole
+
+```cpp
+bool removeRole(Snowflake<Server> serverID, Snowflake<User> userID, Snowflake<Role> roleID);
+```
+
+Takes away a role from a member on a server
+
+#### Parameters
+<table>
+  <tbody>
+      <tr><td><strong>serverID</strong></td>
+        <td>The id of the server with the user you want to remove the role from</td></tr>
+      <tr><td><strong>memberID</strong></td>
+        <td>The id of the user that you want to remove the role from</td></tr>
+      <tr><td><strong>roleID</strong></td>
+        <td>The id of role that you want to remove from the user</td></tr>
+  </tbody>
+</table>
+
+#### Return value
+Returns ``true`` on success
+
+#### Other Details
+[Declared in `client.h`](https://github.com/NoNamer64/sleepy-discord/blob/master/include/sleepy_discord/client.h) and [defined in `endpoints.cpp`](https://github.com/NoNamer64/sleepy-discord/blob/master/include/sleepy_discord/endpoints.cpp)
+
+Uses [Remove Guild Member Role](https://discordapp.com/developers/docs/resources/guild#remove-guild-member-role)
+
+### getRoles
+
+```cpp
+ArrayResponse<Role> getRoles(Snowflake<Server> serverID);
+```
+
+Gets all roles in a server
+
+#### Parameters
+<table>
+  <tbody>
+      <tr><td><strong>serverID</strong></td>
+        <td>The id of the server you want to get the roles from</td></tr>
+  </tbody>
+</table>
+
+#### Return value
+Returns an array of ``Role`` objects
+
+#### Other Details
+[Declared in `client.h`](https://github.com/NoNamer64/sleepy-discord/blob/master/include/sleepy_discord/client.h) and [defined in `endpoints.cpp`](https://github.com/NoNamer64/sleepy-discord/blob/master/include/sleepy_discord/endpoints.cpp)
```

Comment:
404 error, I think you get the point, so I'm not going to keep going without continuing to point this out